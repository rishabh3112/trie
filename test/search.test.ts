import Store from "../src";

describe("search functionality", () => {
  it("should search item which exists in store", async () => {
    const store = new Store<string>();

    await store.add("hinata", "karusuno");
    await store.add("kageyama", "karusuno");

    const hinata = await store.search("hinata");
    const kageyama = await store.search("kageyama");

    expect(hinata).toMatchSnapshot();
    expect(kageyama).toMatchSnapshot();
  });

  it("should return empty if no items found for the name", async () => {
    const store = new Store<string>();

    await store.add("hinata", "karusuno");
    await store.add("kageyama", "karusuno");

    const daichi = await store.search("daichi");

    expect(daichi).toMatchSnapshot();
  });

  it("should return limited items with limit is provided", async () => {
    const store = new Store<string>();

    await store.add("pickachu", "ash");
    await store.add("pichu", "unknown");

    const limitOne = await store.search("pic", 1);

    expect(limitOne).toMatchSnapshot();
  });
});
