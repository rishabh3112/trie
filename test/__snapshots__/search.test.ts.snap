// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search functionality should return empty if no items found for the name 1`] = `Array []`;

exports[`search functionality should return limited items with limit is provided 1`] = `
Array [
  Object {
    "data": "ash",
    "id": "store#8",
    "key": "pickachu",
  },
]
`;

exports[`search functionality should search item which exists in store 1`] = `
Array [
  Object {
    "data": "karusuno",
    "id": "store#6",
    "key": "hinata",
  },
]
`;

exports[`search functionality should search item which exists in store 2`] = `
Array [
  Object {
    "data": "karusuno",
    "id": "store#14",
    "key": "kageyama",
  },
]
`;
